src/core/infra/recommendation.go

```diff
@@ -1222,35 +1222,39 @@ func RecommendVmPerformance(nsId string, specList *[]model.SpecInfo) ([]model.Sp
 
 // RecommendK8sNode is func to recommend a node for K8sCluster
 func RecommendK8sNode(nsId string, plan model.RecommendSpecReq) ([]model.SpecInfo, error) {
-	emptyObjList := []model.SpecInfo{}
-
-	limitOrig := plan.Limit
-	plan.Limit = strconv.Itoa(math.MaxInt)
-
-	SpecInfoListForVm, err := RecommendSpec(nsId, plan)
-	if err != nil {
-		return emptyObjList, err
+	// K8s node minimum requirements
+	const minVCPU = 2
+	const minMemoryGiB = 4.0
+
+	// Add K8s minimum requirements to the filter policy
+	vCPUCondition := model.FilterCondition{
+		Metric: "vCPU",
+		Condition: []model.Operation{
+			{
+				Operand:  strconv.Itoa(minVCPU),
+				Operator: ">=",
+			},
+		},
 	}
 
-	limitNum, err := strconv.Atoi(limitOrig)
-	if err != nil {
-		limitNum = math.MaxInt
+	memoryCondition := model.FilterCondition{
+		Metric: "memoryGiB",
+		Condition: []model.Operation{
+			{
+				Operand:  strconv.FormatFloat(minMemoryGiB, 'f', 1, 64),
+				Operator: ">=",
+			},
+		},
 	}
 
-	SpecInfoListForK8s := []model.SpecInfo{}
-	count := 0
-	for _, SpecInfo := range SpecInfoListForVm {
-		if strings.Contains(SpecInfo.InfraType, model.StrK8s) ||
-			strings.Contains(SpecInfo.InfraType, model.StrKubernetes) {
-			SpecInfoListForK8s = append(SpecInfoListForK8s, SpecInfo)
-			count++
-			if count == limitNum {
-				break
-			}
-		}
+	// Initialize filter policy if not exists
+	if plan.Filter.Policy == nil {
+		plan.Filter.Policy = []model.FilterCondition{}
 	}
 
-	return SpecInfoListForK8s, nil
+	plan.Filter.Policy = append(plan.Filter.Policy, vCPUCondition, memoryCondition)
```

Comment:
@hanizang77 what if, there is existing memoryGiB or/and vCPU conditions already in the request from a user?